Library/Homebrew/formula.rb

```diff
@@ -2005,6 +2005,56 @@ def to_hash
     hsh
   end
 
+  # @private
+  def to_hash
+    hash = to_hash_without_variations
+    variations = {}
+
+    hash_keys_to_skip = %w[installed linked_keg pinned outdated]
+    os_versions = MacOSVersions::SYMBOLS.keys.append :linux
+
+    if path.exist? && self.class.instance_eval { @on_system_blocks_exist }
+      formula_contents = path.read
+      [:arm, :intel].each do |arch|
+        os_versions.each do |os_name|
+          # Big Sur is the first version of macOS that supports arm
```

Comment:
```suggestion
          # Big Sur is the first version of macOS that runs on ARM
```

Might be nice to turn this into a method/code somewhere?

Reply:
Sorry, not sure I know what you mean here

Reply:
A new `MacOS::Version#supports_arm?` or something?

Reply:
Gotcha. I pushed a slightly different change but I think it accomplishes the same thing. See if that's what you were looking for.

It also modifies the `variations` keys to match what `Utils::Bottles.tag` would return to simplify the merging